### PR TITLE
dev-util/hip: QA fixes

### PR DESCRIPTION
**Pull request by littlewu2508**
Signed-off-by: Yiyang Wu <xgreenlandforwyy@gmail.com>

### CODE REVIEW

File: dev-util/hip/hip-5.1.3-r1.ebuild [hunk 1]
```diff
@@ -81,12 +81,14 @@ src_prepare() {
...
+		-e "/${HIP_COMMON_DIR}\/cmake DESTINATION/d" -i CMakeLists.txt || die
 	sed -e "/LIBRARY DESTINATION/s:lib:$(get_libdir):" -i src/CMakeLists.txt || die
 
 	sed -e "/\.hip/d" \
 		-e "s,DESTINATION lib,DESTINATION $(get_libdir),g" \
-		-e "/\(cmake\|samples\)/s,DESTINATION \.,DESTINATION share,g" \
+		-e "/samples/s,DESTINATION \.,DESTINATION share/hip,g" \
+		-e "/cmake DESTINATION/d" \
+		-e "/DESTINATION docs/s,docs,\${CMAKE_INSTALL_DOCDIR},g" \
 		-e "/CPACK_RESOURCE_FILE_LICENSE/d" -i packaging/CMakeLists.txt || die
```
Comment (heroxbd) [inline dev-util/hip/hip-5.1.3-r1.ebuild:92]: What are these?  They are not documented in the commit messages.

File: dev-util/hip/hip-5.1.3-r1.ebuild [hunk 2]
```diff
@@ -81,12 +81,14 @@ src_prepare() {
...
 	# correct libs and cmake install dir
 	sed -e "/LIB_INSTALL_DIR/s:PREFIX}/lib:PREFIX}/$(get_libdir):" \
-		-e "/${HIP_COMMON_DIR}\/cmake DESTINATION/s: .): $(get_libdir)):" -i CMakeLists.txt || die
+		-e "/\${HIP_COMMON_DIR}/s:cmake DESTINATION .):cmake/ DESTINATION share/cmake/Modules):" -i CMakeLists.txt || die
 	sed -e "/LIBRARY DESTINATION/s:lib:$(get_libdir):" -i src/CMakeLists.txt || die
 
 	sed -e "/\.hip/d" \
 		-e "s,DESTINATION lib,DESTINATION $(get_libdir),g" \
-		-e "/\(cmake\|samples\)/s,DESTINATION \.,DESTINATION share,g" \
+		-e "/samples/s,DESTINATION \.,DESTINATION share/hip,g" \
```
Comment (heroxbd) [inline dev-util/hip/hip-5.1.3-r1.ebuild:89]: The samples belong to `/usr/share/doc/${P}/`.

Comment (littlewu2508) [reply dev-util/hip/hip-5.1.3-r1.ebuild]: Actually it is `/usr/share/doc/${PF}/`

**Comment by littlewu2508** [comment]
I pushed two more commits -- fixing sample install location using patch, and add `doc` USE flag to control document installing. If looking good, shall I squash them?

**Comment by heroxbd** [comment]
littlewu2508 ***@***.***> writes:

> Actually it is /usr/share/doc/${PF}/

Yes, you are correct.

File: dev-util/hip/hip-5.1.3-r1.ebuild [hunk 3]
```diff
@@ -4,8 +4,10 @@
 EAPI=8
 
 PYTHON_COMPAT=( python3_{8..11} )
+DOCS_BUILDER="doxygen"
+DOCS_DEPEND="media-gfx/graphviz"
 
-inherit cmake llvm prefix python-any-r1
+inherit cmake docs llvm prefix python-any-r1
```
Comment (heroxbd) [inline dev-util/hip/hip-5.1.3-r1.ebuild:10]: The convention of Gentoo is to use `doc`.

```
equery h doc
```

File: dev-util/hip/hip-5.1.3-r1.ebuild [hunk 4]
```diff
@@ -4,8 +4,10 @@
...
 
 LLVM_MAX_SLOT=14
```
Comment (heroxbd) [inline dev-util/hip/hip-5.1.3-r1.ebuild:12]: What's this?

File: dev-util/hip/hip-5.1.3-r1.ebuild (hunk 3 above)
Comment (littlewu2508) [reply dev-util/hip/hip-5.1.3-r1.ebuild:10]: Sorry, I can't get your point. FWICS there is only docs.eclass, not doc.eclass

Comment (heroxbd) [reply dev-util/hip/hip-5.1.3-r1.ebuild:10]: My mistake. Never mind.

File: dev-util/hip/hip-5.1.3-r1.ebuild (hunk 4 above)
Comment (littlewu2508) [reply dev-util/hip/hip-5.1.3-r1.ebuild:12]: I followed the example of `llvm.eclass`, which recognize `LLVM_MAX_SLOT`.

Comment (littlewu2508) [reply dev-util/hip/hip-5.1.3-r1.ebuild:12]: Now the only usable llvm/clang SLOT is 14 for ROCm toolchain. In the future, we can unset this variable, and specify DEPEND as `>=sys-devel/clang-14.0.6-r1`

File: dev-util/hip/hip-5.1.3-r1.ebuild [hunk 5]
```diff
@@ -23,7 +25,7 @@ KEYWORDS="~amd64"
 LICENSE="MIT"
 SLOT="0/$(ver_cut 1-2)"
 
-IUSE="debug profile"
+IUSE="debug doc profile"
```
Comment (AndrewAmmerlaan) [inline dev-util/hip/hip-5.1.3-r1.ebuild:28]: the doc flag is already set by `docs.eclass`

File: dev-util/hip/hip-5.1.3-r1.ebuild [hunk 6]
```diff
@@ -150,6 +156,11 @@ src_configure() {
 	cmake_src_configure
 }
 
+src_compile() {
+	HIP_PATH=${HIP_S} docs_compile
+	cmake_src_compile
```
Comment (AndrewAmmerlaan) [inline dev-util/hip/hip-5.1.3-r1.ebuild:161]: to actually compile the docs we need to call `docs_compile` somewhere in the compile phase

File: dev-util/hip/hip-5.1.3-r1.ebuild (hunk 5 above)
Comment (littlewu2508) [reply dev-util/hip/hip-5.1.3-r1.ebuild:28]: Got it

**Pull request closed without merging**